addons/bus/static/tests/bus_tests.js

```diff
@@ -423,6 +423,66 @@ QUnit.module('Bus', {
         await updateLastNotificationDeferred;
         assert.verifySteps([`initialize_connection - 0`]);
     });
-});
 
+    QUnit.test('Websocket connection is refreshed upon user log in', async function (assert) {
+        const connectionInitializedDeferred = makeDeferred();
+        const disconnectionDoneDeferred = makeDeferred();
+        patchWebsocketWorkerWithCleanup({
+            _initializeConnection(client, data) {
+                this._super(client, data);
+                connectionInitializedDeferred.resolve();
+            },
+        });
+
+        // first tab connects with current user as uid.
+        const firstTabEnv = await makeTestEnv();
+        firstTabEnv.services['bus_service'].addEventListener('disconnect', ({ detail }) => {
+            assert.step(`disconnect - ${detail.code}`);
+            disconnectionDoneDeferred.resolve();
+        });
+        await connectionInitializedDeferred;
+
+        // second tab connects to the worker after a disconnection: uid
+        // is now undefined.
+        patchWithCleanup(session, {
+            uid: undefined,
+        });
+        await makeTestEnv();
+        await disconnectionDoneDeferred;
+
+        assert.verifySteps([`disconnect - ${WEBSOCKET_CLOSE_CODES.SESSION_EXPIRED}`]);
+    });
+
+    QUnit.test('Websocket connection is refreshed upon user log out', async function (assert) {
```

Comment:
name of tests are inverted to what is done inside

Reply:
updated :+1: